src/main/java/org/jabref/gui/preferences/EntryEditorTabViewModel.java

```diff
@@ -33,37 +32,42 @@
     private final BooleanProperty firstNameModeFullProperty = new SimpleBooleanProperty();
     private final BooleanProperty firstNameModeBothProperty = new SimpleBooleanProperty();
 
-    private AutoCompletePreferences autoCompletePreferences;
-
     private final DialogService dialogService;
-    private final JabRefPreferences preferences;
+    private final PreferencesService preferencesService;
+    private final EntryEditorPreferences initialEntryEditorPreferences;
+    private final AutoCompletePreferences initialAutoCompletePreferences;
 
-    public EntryEditorTabViewModel(DialogService dialogService, JabRefPreferences preferences) {
+    public EntryEditorTabViewModel(DialogService dialogService, PreferencesService preferencesService) {
         this.dialogService = dialogService;
-        this.preferences = preferences;
-        this.autoCompletePreferences = preferences.getAutoCompletePreferences();
+        this.preferencesService = preferencesService;
+        this.initialEntryEditorPreferences = preferencesService.getEntryEditorPreferences();
+        this.initialAutoCompletePreferences = preferencesService.getAutoCompletePreferences();
     }
 
     @Override
     public void setValues() {
-        openOnNewEntryProperty.setValue(preferences.getBoolean(JabRefPreferences.AUTO_OPEN_FORM));
-        defaultSourceProperty.setValue(preferences.getBoolean(JabRefPreferences.DEFAULT_SHOW_SOURCE));
-        enableRelatedArticlesTabProperty.setValue(preferences.getBoolean(JabRefPreferences.SHOW_RECOMMENDATIONS));
-        acceptRecommendationsProperty.setValue(preferences.getBoolean(JabRefPreferences.ACCEPT_RECOMMENDATIONS));
-        enableLatexCitationsTabProperty.setValue(preferences.getBoolean(JabRefPreferences.SHOW_LATEX_CITATIONS));
-        enableValidationProperty.setValue(preferences.getBoolean(JabRefPreferences.VALIDATE_IN_ENTRY_EDITOR));
-        enableAutoCompleteProperty.setValue(autoCompletePreferences.shouldAutoComplete());
-        autoCompleteFieldsProperty.setValue(autoCompletePreferences.getCompleteNamesAsString());
-
-        if (autoCompletePreferences.getOnlyCompleteFirstLast()) {
+        // ToDo: Include EntryEditorFieldsEditor in PreferencesDialog
```

Comment:
Is this todo still relevant?

Reply:
Had the wrong name. Should be CustomizeGeneralFieldsDialog